### PR TITLE
Enable Server-Side Apply mode by default

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: provider/pkg/provider/provider.go [hunk 1]
```diff
@@ -442,8 +442,8 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 		if enabled, exists := os.LookupEnv("PULUMI_K8S_ENABLE_SERVER_SIDE_APPLY"); exists {
 			return enabled == trueStr
 		}
-		// Default to false.
-		return false
+		// Default to true.
```
[the remark this answers is not in the document]
Comment (squaremo) [reply provider/pkg/provider/provider.go:445]: This func is locally scoped, invoked immediately, and never mentioned again. Why not just run the code within? Or at least factor out the pattern of checking the vars, then the environment, then defaulting, and use that for all of these settings.

File: provider/pkg/gen/schema.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 					DeprecationMessage: "This option has been replaced by `enableServerSideApply`.",
 				},
 				"enableServerSideApply": {
-					Description: "BETA FEATURE - If present and set to true, enable Server-Side Apply mode.\nSee https://github.com/pulumi/pulumi-kubernetes/issues/2011 for additional details.\nThis feature is in developer preview, and is disabled by default.",
+					Description: "If present and set to true, enable Server-Side Apply mode.\nSee https://github.com/pulumi/pulumi-kubernetes/issues/2011 for additional details.",
```
Comment (squaremo) [inline provider/pkg/gen/schema.go:81]: Shouldn't this be the other way around, if it defaults to true? As in,

> If present at set to false, Server-Side Apply mode is disabled. [etc...]

**Review by squaremo — COMMENTED** [review]
Does the comment in provider.go starting L2138 need to be updated, now the default is server-side apply? It says

> In the next major release, we will default to using Server-side Apply, which will simplify this logic.

File: provider/pkg/provider/provider.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lblackstone) [reply provider/pkg/provider/provider.go:445]: Yep, this code is overdue for some cleanup. It used to just be one option checked like that, and got copy-pasted a bunch of times. :)

File: provider/pkg/gen/schema.go (hunk 2 above)
Comment (lblackstone) [reply provider/pkg/gen/schema.go:81]: Good catch, fixed!

**Comment by lblackstone** [comment]
> Does the comment in provider.go starting L2138 need to be updated, now the default is server-side apply? It says
> 
> > In the next major release, we will default to using Server-side Apply, which will simplify this logic.

We plan to drop the provider flag in `v4`, but may need to maintain some of the existing client-side logic for previews, etc. This would be an implementation detail rather than an opt-in, though. I'll leave this comment for now.

**Review by viveklak — APPROVED** [review]

**Comment by dotansimha** [comment]
> > "message": "error: Preview failed: 1 error occurred:\n\t* the Kubernetes API server reported that "cert-manager-webhook" failed to fully initialize or become live: use `pulumi.com/patchForce` to override the conflict: Apply failed with 2 conflicts: conflicts with

Same here 👍 we are getting this with cert-manager deployment:

```
kubernetes:admissionregistration.k8s.io/v1:ValidatingWebhookConfiguration (cert-manager-webhook):
      error: 1 error occurred:
      	* the Kubernetes API server reported that "cert-manager-webhook" failed to fully initialize or become live: use `pulumi.com/patchForce` to override the conflict: Apply failed with 2 conflicts: conflicts with "admissionsenforcer" using admissionregistration.k8s.io/v1:
      - .webhooks[name="webhook.cert-manager.io"].namespaceSelector
      conflicts with "pulumi-resource-kubernetes" using admissionregistration.k8s.io/v1:
      - .webhooks[name="webhook.cert-manager.io"].rules
      - ```

**Comment by squaremo** [comment]
@pstovik @dotansimha Sorry about that folks. SSA being on by default was reverted in v3.22.1, until we can clear up problems like those you reported. Bumping the package version will get the new provider (e.g., `npm update @pulumi/kubernetes`).

**Pull request merged**